pkg/namespace/consts.go

```diff
@@ -36,39 +36,50 @@ var (
 	// NamespaceVersionZeroPrefix is the prefix of a namespace ID for version 0.
 	NamespaceVersionZeroPrefix = bytes.Repeat([]byte{0}, NamespaceVersionZeroPrefixSize)
 
-	// TxNamespace is the namespace reserved for transaction data.
-	TxNamespace = MustNewV0([]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 1})
+	// TxNamespace is the namespace reserved for ordinary Cosmos SDK transactions.
+	TxNamespace = primaryReservedNamespace(0x01)
 
 	// IntermediateStateRootsNamespace is the namespace reserved for
 	// intermediate state root data.
-	IntermediateStateRootsNamespace = MustNewV0([]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 2})
+	IntermediateStateRootsNamespace = primaryReservedNamespace(0x02)
 
 	// PayForBlobNamespace is the namespace reserved for PayForBlobs transactions.
-	PayForBlobNamespace = MustNewV0([]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 4})
+	PayForBlobNamespace = primaryReservedNamespace(0x04)
 
-	// ReservedPaddingNamespace is the namespace used for padding after all
-	// reserved namespaces. In practice this padding is after transactions
-	// (ordinary and PFBs) but before blobs.
-	ReservedPaddingNamespace = MustNewV0([]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 255})
+	// PrimaryReservedPaddingNamespace is the namespace used for padding after all
+	// primary reserved namespaces.
+	PrimaryReservedPaddingNamespace = primaryReservedNamespace(0xFF)
```

Comment:
[optional] note to reviewers, I think it may be worth including one extra breaking change to this PR. Specifically:
 
```suggestion
	PrimaryReservedPaddingNamespace = primaryReservedNamespace(0xFE)
```

This change would differentiate MaxPrimaryReserved and PrimaryReservedPadding which seems redundant.

This proposal is partially motivated by the fact that tail padding is 0xFE which leaves one extra namespace at the very end for parity shares. I wonder if it may be helpful to have one extra namespace at the end of the primary range just in case.

Reply:
[Question] Does this mean having one padding type for the primary reserved namespaces, and another padding type for the secondary ones? not the same padding namespace?

Reply:
Correct. Multiple namespaces are already used for different padding shares. See https://celestiaorg.github.io/celestia-app/specs/shares.html#reserved-padding-share